code/client/clrcore/External/WeaponComponent.cs

```diff
@@ -109,328 +116,396 @@ public static string GetComponentDisplayNameFromHash(WeaponHash hash, WeaponComp
 		[SecurityCritical]
 		private static string _GetComponentDisplayNameFromHash(WeaponHash hash, WeaponComponentHash component)
 		{
-			if (hash == WeaponHash.KnuckleDuster)
+			switch (component)
 			{
-				switch (component)
```

Comment:
what's the reason for this big of a change?

Reply:
I thought it would be a good idea to adjust it according to what ordering my loop would give them out on, not a good idea after all...

Reply:
Nevermind, I honestly have no idea why I did it this way lol

Reply:
I actually see why now, lots of attachments were added and I added some new entries in between the others to have some kind of order, the original order was kept. I also 'merged' some cases with each other because otherwise you'd have duplicate things like the attached screenshot
![image](https://user-images.githubusercontent.com/82737367/209408554-3f9d4003-5dbf-47ae-9cd8-6f542c0c7622.png)

Reply:
The order is actually mostly the same, it's just that the gxt entries have more components tied to them and I added the knuckleduster to the list so it went a nesting down if you know what I mean